### DISCUSSION TITLE
Product Stock

### DISCUSSION THREAD

**Discussion by ardvz** [Paymenter/Paymenter#512] [Ideas]
Add stock check feature to pterodactyl extension. So the system will check the remaining memory on the nodes on the pterodactyl panel and calculate with the product displayed.

For example, if only 10GB of memory remains on the node, then the product display will only activate purchases for 1GB - 10GB products and so on. Maybe to deal with lots of nodes in one location, you can create something like forEach so that the system counts each node, not all of them.

**Comment by ItzFLKN** [comment]
That could be cool,  or query all nodes and calculate the absolute max you can free up if you load smaller servers into the gaps on nodes and free up extra space for a big server to jump onto one node?  don't know if kubernetes is required for that tho, rather than just docker as it can just failover (I think) to another node with little to no downtime?

**Comment by CorwinDev** [comment]
https://builtbybit.com/resources/ptero-auto-stock-updater-for-paymenter.70768/?ref=398830
